File: app/src/main/java/com/nytsample/android/presenter/NewsListPresenter.java
package com.nytsample.android.presenter;

import com.nytsample.android.data.api.NYTimesAPI;
import com.nytsample.android.data.model.NewsFeedResponseModel;
import com.nytsample.android.view.def.NewsListView;

import org.jetbrains.annotations.NotNull;

public class NewsListPresenter {
    private final NewsListView view;

    public NewsListPresenter(NewsListView view) {
        this.view = view;
    }

    public void fetchNews(int offset) {
        view.showLoading();

        NYTimesAPI.getNewsFeed(offset, new NYTimesAPI.NewsFeedCallback() {
            @Override
            public void onSuccess(@NotNull NewsFeedResponseModel newsFeedResponseModel) {
                view.showResults(newsFeedResponseModel);
            }

            @Override
            public void onFailure(@NotNull String error) {
                view.showError(error);
            }
        });
    }
}

File: app/src/main/java/com/nytsample/android/presenter/NewsListPresenter.kt
package com.nytsample.android.presenter

import com.nytsample.android.data.api.NYTimesApi
import com.nytsample.android.data.model.NewsFeedResponseModel
import com.nytsample.android.view.def.NewsListView
import com.nytsample.android.data.api.NYTimesApi.Companion.NewsFeedCallback

class NewsListPresenter(val view: NewsListView) {
    fun fetchNews(offset: Int = 0) {
        view.showLoading()
        NYTimesApi.getNewsFeed(offset, object : NewsFeedCallback {
            override fun onSuccess(newsFeedResponseModel: NewsFeedResponseModel) {
                view.showResults(newsFeedResponseModel)
            }

            override fun onFailure(error: String) {
                view.showError(error)
            }
        })
    }
}